tests/utils/test_dates.py

```diff
@@ -27,18 +28,19 @@
 from airflow.utils import dates, timezone
 
 
-class TestDates(unittest.TestCase):
-    def test_days_ago(self):
-        today = pendulum.today()
-        today_midnight = pendulum.instance(datetime.fromordinal(today.date().toordinal()))
+class TestDates:
+    @pytest.mark.parametrize('timezone', ['UTC', 'America/Los_Angeles'])
+    def test_days_ago(self, timezone):
+        with mock.patch('airflow.utils.timezone.TIMEZONE', pendulum.tz.timezone(timezone)):
+            today_midnight = pendulum.today(timezone)
 
-        assert dates.days_ago(0) == today_midnight
-        assert dates.days_ago(100) == today_midnight + timedelta(days=-100)
-
-        assert dates.days_ago(0, hour=3) == today_midnight + timedelta(hours=3)
-        assert dates.days_ago(0, minute=3) == today_midnight + timedelta(minutes=3)
-        assert dates.days_ago(0, second=3) == today_midnight + timedelta(seconds=3)
-        assert dates.days_ago(0, microsecond=3) == today_midnight + timedelta(microseconds=3)
+            assert today_midnight.tzinfo == dates.days_ago(2).tzinfo == pendulum.tz.timezone(timezone)
+            assert dates.days_ago(0) == today_midnight
+            assert dates.days_ago(100) == today_midnight.add(days=-100)  # For the DST boundary test.
+            assert dates.days_ago(0, hour=3) == today_midnight + timedelta(hours=3)
+            assert dates.days_ago(0, minute=3) == today_midnight + timedelta(minutes=3)
+            assert dates.days_ago(0, second=3) == today_midnight + timedelta(seconds=3)
+            assert dates.days_ago(0, microsecond=3) == today_midnight + timedelta(microseconds=3)
```

Comment:
These indentation changes don’t seem to be necessary?

Reply:
Those indentation is necessary because `dates.days_ago` uses `timezone.TIMEZONE`. So it needs to use patched `timezone.TIMEZONE` by `with mock.patch()` statement.
```python
# days_ago()
today = datetime.now(timezone.TIMEZONE).replace(
    hour=hour, minute=minute, second=second, microsecond=microsecond
)
```